### DISCUSSION TITLE
Upload any file

### DISCUSSION THREAD

**Discussion by evgeniyworkbel** [facebook/lexical#4181] [Q&A]
Is there an opportunity to upload any file like music, video, txt, pdf, zip, epub and etc? I didn't find info about it at all. As I saw, I only can insert image, gif, youtube video

**Comment by ivailop7** [comment]
In terms of the uploading, you can upload any file, it's more about what do you do with the data afterwards.

In terms of rendering, for the some of the HTML elements, you can extend the Element Node, for example the audio/video you can render the HTML5 tag for them in the createDOM. For the more complicated use cases - eg. PDF rendering, you can extend the Decorator Node, which you can think of is like an iframe, where you can embedded anything basically.
